pkg/api/v1/types.go

```diff
@@ -2131,6 +2131,11 @@ type PodAffinityTerm struct {
 	// for affinity and for RequiredDuringScheduling pod anti-affinity, empty topologyKey is not allowed.
 	// +optional
 	TopologyKey string `json:"topologyKey,omitempty" protobuf:"bytes,3,opt,name=topologyKey"`
+	// MinMatchingPods specifies the minimal number of pod that matching the selector; the term is considered to
```

Comment:
I found the description somewhat confusing. Also, I'm not sure MinMatchingPods is a good name.

The meaning is a little different when it is used with affinity vs. anti-affinity, so I would suggest to write separate comments for each use case.

I didn't look at the implementation yet, but here is what it should be implementing:

```
When used with anti-affinity, a node N is eligible only if the number of pods
matching the LabelSelector that are running on nodes with the same value
for key TopologyKey as N has is less than MinMatchingPods. For example,
if MinMatchingPods is 1 (the default), then after this pod is scheduled to node M,
there will not be more than one pod from the set selected by the LabelSelector
running on nodes with M's value for key TopologyKey.

When used with affinity, a node N is eligible only if the number of pods
matching the LabelSelector that are running on nodes with the same value
for key TopologyKey as N has is greater or equal to MinMatchingPods.
For example, if MinMatchingPods is 1 (the default), then after this pod is
scheduled to node M, there will be at least 2 pods from the set selected by
the LabelSelector running on nodes with M's value for key TopologyKey.
```

(BTW, I actually don't think we need to bother to make this apply to affinity -- the main use case is with anti-affinity. But if it's easier to implement both affinity and anti-affinity, that's OK.)

I would suggest calling it "NumMatchingPods" or something like that. there isn't really a good variable name that captures the full meaning, and anyway this feature is complicated enough that the user is going to need to read the description or documentation anyway.

Please update the comment, change the variable name, and verify that it is implementing what I described above.